opt_einsum/paths.py

```diff
@@ -43,6 +43,18 @@ def __call__(self, inputs, output, size_dict, memory_limit=None):
 
     where ``path`` is a list of int-tuples specifiying a contraction order.
     """
+
+    def check_args_against_first_call(self, inputs, output, size_dict):
+        """Utility that stateful optimizers can use to ensure they are not
+        called with different contractions across separate runs.
+        """
+        args = (inputs, output, size_dict)
+        if not hasattr(self, '_first_call_args'):
```

Comment:
Initializing this to `None` might be simpler.

```
if self._first_call_args is not None:
```

Reply:
It would be preferable, the only reason I haven't is that ``PathOptimizer`` itself doesn't have an ``__init__`` so there is no single place to initialize the ``_first_call_args``.

Reply:
Good point, can you add a comment to this effect?

Reply:
Have done, let me know if it needs more clarification!